core/src/main/java/io/dekorate/processor/AbstractAnnotationProcessor.java

```diff
@@ -114,7 +116,15 @@ public synchronized void init(ProcessingEnvironment processingEnv) {
         throw DekorateException.launderThrowable(e);
       } 
     }
-    return Collections.emptyMap();
+    return result;
+  }
+
+  private void mergeKubernetesProperties(Map<String, Object> result, Map<String, Object> newProps) {
+    if(newProps.containsKey("kubernetes") && result.containsKey("kubernetes")) {
```

Comment:
Why are we checking if both maps contain the key `kubernetes`?

I'd like at this point to make no assumptions on they keys as it would be nice if this feature also worked for `openshift` and `component` keys too.

Reply:
Because if you do `result.putAll(newProps)` even if `newProps` has new `kubernetes` properties everything will get overridden because the property `kubernetes` exists in both `maps` even though the properties under the `map` are different.  So essentially it is a very basic merge functionality.

If there are other keys we should worry about besides `kubernetes` then yes I agree we should add logic for that as well.

I can probably generalize the logic to work for any key.